### PR TITLE
Adding Dynamic features

### PR DESCRIPTION
**Pull request by mrm9084**

### CODE REVIEW

**Review by saragluna — APPROVED** [review]

**Review by saragluna — APPROVED** [review]
lgtm

File: docs/src/main/asciidoc/feature-management.adoc [hunk 1]
```diff
@@ -62,7 +62,16 @@ feature-management:
...
 * `feature-u` is used in conjunction with Feature Filters. These Filters will be defined under the `enabled-for` property.  In this case, `feature-u` has one Feature Filter called `Random`, which does not require any configuration, so only the name property is required. 
 * `feature-v` specifies a Feature Filter named `TimeWindowFilter`. This Feature Filter can be passed parameters to use as configuration. In this case, for a `TimeWindowFilter`, pass in the start and end times for which the feature will be active.
-* `feature-w` is used for the `AlwaysOnFilter`, which will always evaluate to `true`. The `evaluate` field is used to stop the evaluation of the Feature Filters, and results in the Feature Filter always return `false`.
+* `feature-w` is used for the `AlwaysOnFilter`, which will always evaluate to `true`. The `evaluate` field is used to stop the evaluation of the Feature Filters, and results in the Feature Filter always return `false`.
+
+==== Feature Variants
+
+Feature Variants are multiple options of a feature that can be targeted at different groups of users. A Feature Variant is defined by four values:
+
+* name - Identifier for the varieant.
```
Comment (AlexandraKemperMS) [inline docs/src/main/asciidoc/feature-management.adoc:71]: variant

File: docs/src/main/asciidoc/feature-management.adoc [hunk 2]
```diff
@@ -62,7 +62,16 @@ feature-management:
...
+* default - Indicates which variant should be used if no variant is assigned. In each set, only one can be marked as default.
+* configuration-reference - A pointer to which `@ConfigurationProperties` are used for that variant.
```
Comment (AlexandraKemperMS) [inline docs/src/main/asciidoc/feature-management.adoc:73]: Could you add here that there is no limit to how many properties are associated with a variant? That would be my next question when reading this - what are the limitations, etc.

File: docs/src/main/asciidoc/feature-management.adoc [hunk 3]
```diff
@@ -285,4 +294,118 @@ Options are available to customize how targeting evaluation is performed across
     public TargetingFilter targetingFilter(ITargetingContextAccessor contextAccessor) {
         return new TargetingFilter(contextAccessor, new TargetingEvaluationOptions().setIgnoreCase(true));
     }
-----
\ No newline at end of file
+----
+
+=== Evaluating Dynamic Features
+
+Dynamic Features are enabled through the creation <<Feature Variants, feature variants>>, where each variant describes who is assigned that variant, and a reference to the configurations that make that variant. DynamicFeatures are accessed through the `DynamicFeatureManager`.
```
Comment (AlexandraKemperMS) [inline docs/src/main/asciidoc/feature-management.adoc:301]: I would reiterate here that a variant is one of multiple options for a feature to make sure people understand the full picture.

**Pull request closed without merging**